### PR TITLE
[DEVHUB-1167] - Make secondary nav sticky instead of consistent nav, lock background scrolling on overlays.

### PR DESCRIPTION
**Pull request by mcolella14**
[DEVHUB-1167](https://jira.mongodb.org/browse/DEVHUB-1167)

The first commit was just making the secondary nav sticky over the consistent nav. 

The second commit centers around an issue I noticed where on the mobile nav menu (as well as on the mobile search filter overlay), if a user scrolls to the bottom of the overlay, they can keep scrolling the page in the background. This is just a weird experience, so I added a context value so we have some global knowledge of whether an overlay is present. We then lock the main body from scrolling if this is the case.

### CODE REVIEW

File: src/components/seconardynavnew/mobile-styles.ts [hunk 1]
```diff
@@ -1,11 +1,23 @@
 import { ThemeUIStyleObject } from 'theme-ui';
 
+export const navWrapperStyles: ThemeUIStyleObject = {
+    boxSizing: 'border-box',
+    background: '#fff',
+    backgroundColor: 'white',
```
Comment (justinmckenzie) [inline src/components/seconardynavnew/mobile-styles.ts:6]: why `background` and `backgroundColor`?

File: src/components/seconardynavnew/mobile-styles.ts [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
+    display: ['block', 'block', 'block', 'none'],
```
Comment (justinmckenzie) [inline src/components/seconardynavnew/mobile-styles.ts:7]: `display: ['block', null, null, 'none']` (unless im misunderstanding that on my end)

**Review by justinmckenzie — APPROVED** [review]
Nitpicks from existing CSS, everything else looks great

File: src/components/seconardynavnew/mobile-styles.ts (hunk 1 above)
Comment (mcolella14) [reply src/components/seconardynavnew/mobile-styles.ts:6]: no idea lol, I'll correct this

File: src/components/seconardynavnew/mobile-styles.ts (hunk 2 above)
Comment (mcolella14) [reply src/components/seconardynavnew/mobile-styles.ts:7]: ah yep good catch!

**Review by gfunk1230 — APPROVED** [review]

**Review by justinmckenzie — APPROVED** [review]

**Pull request merged**